web/src/TestgridIndex.ts

```diff
@@ -1,63 +1,225 @@
+                /\W/g,
+                ''
+              )}/tabs"
+            >
+              <div class="container">
+                <p>${dash}</p>
+              </div>
+            </a>
+          </mwc-list-item>
+        `
+      )}
+    </mwc-list>
+  </div>
+`;
 
 @customElement('testgrid-index')
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 export class TestgridIndex extends LitElement {
-  @property({ type: String }) title: string = 'My app';
-
   @property({ type: Array<string> }) dashboards: Array<string> = [];
 
+  @property({ type: Array<string> }) dashboardGroups: Array<string> = [];
+
+  @property({ type: Array<string> }) respectiveDashboards: Array<string> = [];
+
+  @property({ type: Boolean }) show = true;
+
   // TODO(chases2): inject an APIClient object so we can inject it into tests/storybook later
 
   render() {
     return html`
-      <mwc-list>
-        ${map(this.dashboards, (dash: string, index: number) => {
-          if (index !== 0) {
-            return html`
-              <li divider role="separator"></li>
-              <mwc-list-item>${dash}</mwc-list-item>
-            `;
-          }
-          return html`<mwc-list-item>${dash}</mwc-list-item>`;
-        })}
-      </mwc-list>
-      <mwc-button raised @click="${this.getDashboards}">Call API</mwc-button>
+      <mwc-button raised @click="${this.callAPI}">Call API</mwc-button>
+
+      <div class="flex-container">
+        <!-- loading dashboard groups -->
+        <mwc-list style="min-width: 760px">
+          ${map(
+            this.dashboardGroups,
+            (dash: string, index: number) => html`
+              <mwc-list-item
+                id=${index}
+                class="column card dashboard-group"
+                raised
+                @click="${() => this.getRespectiveDashboards(dash)}"
+              >
+                <div class="container">
+                  <p>${dash}</p>
+                </div>
+              </mwc-list-item>
+            `
+          )}
+        </mwc-list>
+
+        <!-- loading dashboards -->
+        ${this.show ? dashboardTemplate(this.dashboards) : ''}
+
+        <!-- loading respective dashboards -->
+        ${!this.show ? dashboardTemplate(this.respectiveDashboards) : ''}
+        ${!this.show
+          ? html`
+              <mwc-button
+                class="column"
+                raised
+                @click="${() => {
+                  this.show = !this.show;
+                }}"
+                >X</mwc-button
+              >
+            `
+          : ''}
+      </div>
     `;
   }
 
+  // function to get dashboards
   getDashboards() {
     this.dashboards = ['Loading...'];
 
-    fetch('http://localhost:8080/api/v1/dashboards').then(async response => {
-      const resp = ListDashboardResponse.fromJson(await response.json());
+    fetch('http://testgrid-data.k8s.io/api/v1/dashboards').then(
```

Comment:
Please pull these hostnames into a constant, as with the API Client object.

That will make it easy to switch between local host and another API instance.

Reply:
Done!